airflow/ti_deps/deps/trigger_rule_dep.py

```diff
@@ -102,12 +102,124 @@ def _get_dep_statuses(
         if ti.task.trigger_rule == TR.ALWAYS:
             yield self._passing_status(reason="The task had a always trigger rule set.")
             return
+        if not ti.task.is_teardown:
+            # a teardown cannot have any indirect setups
+            relevant_setups = {t.task_id: t for t in ti.task.get_upstreams_only_setups()}
+            if relevant_setups:
+                for status, changed in self._evaluate_setup_constraint(
+                    ti=ti, relevant_setups=relevant_setups, dep_context=dep_context, session=session
+                ):
+                    yield status
+                    if status.passed is False and changed:
```

Comment:
I don't think we have None values in passed:
```suggestion
                    if not status.passed and changed:
```

Reply:
Any reason this is resolved without a comment?

Reply:
I accepted the suggestion but then mypy complained so I reverted it

Reply:
Fundamentally there's nothing wrong with saying is false when that's really what you're looking for

Reply:
Cool

Reply:
i guess mypy mighta been unrelated